src/main/java/codeu/controller/BotServlet.java

```diff
@@ -0,0 +1,148 @@
+package codeu.controller;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+public class BotServlet extends HttpServlet {
+
+    private ConversationStore conversationStore;
+    private MessageStore messageStore;
+    private UserStore userStore;
+
+    /**
+     * Sets the {@link ConversationStore} used by this servlet. This function provides a common setup method
+     * for use by the test framework or the servlet's init() function.
+     */
+    void setConversationStore(ConversationStore conversationStore) {
+        this.conversationStore = conversationStore;
+    }
+
+    /**
+     * Sets the {@link MessageStore} used by this servlet. This function provides a common setup method
+     * for use by the test framework or the servlet's init() function.
+     */
+    void setMessageStore(MessageStore messageStore) {
+        this.messageStore = messageStore;
+    }
+
+    /**
+     * Sets the {@link UserStore} used by this servlet. This function provides a common setup method
+     * for use by the test framework or the servlet's init() function.
+     */
+    void setUserStore(UserStore userStore) {
+        this.userStore = userStore;
+    }
+
+    /**
+     * Set up state for handling chat requests.
+     */
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        setConversationStore(ConversationStore.getInstance());
+        setMessageStore(MessageStore.getInstance());
+        setUserStore(UserStore.getInstance());
+    }
+
+    /**
+     * This function fires when a user navigates to the chatbot page. It gets the username from the session
+     * and loads the corresponding chatbot conversation for the user, if the user is logged in, and forwards the
+     * messages in the conversation to chat.jsp
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        String user = (String) request.getSession().getAttribute("user");
+
+        if (user == null) {
+            response.sendRedirect("/login");
+            return;
+        }
+
+        final String conversationTitle = "chatbot-" + user.hashCode();
+
+        Conversation conversation = conversationStore.getConversationWithTitle(conversationTitle);
+        if (conversation == null) {
+            System.out.println("BotServlet-doGet: Chatbot conversation was null for user `" + user + "`: " + conversationTitle);
```

Comment:
For server side code, the usual pattern is to use some Logger to log "INFO" messages since normally servers do not have screens attached to them.
But again, in this case, it is not a big deal. Just FYI